### PR TITLE
DOC: list supported float dtypes I

### LINKED ISSUE #50216: DOC: Supported Float dypes not documented
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/user_guide/basics.html#dtypes

### Documentation problem

In the [dtypes section](https://pandas.pydata.org/docs/user_guide/basics.html#dtypes) a list of integer types are given, but not floating point types. According to [`pandas.to_numeric()`](https://pandas.pydata.org/docs/reference/api/pandas.to_numeric.html#pandas.to_numeric) under the `downcast` argument, it appears `float32` is the smallest floating point type, which implies `float16` is not documented.

In addition, the integer types are listed with capitalization, but it appears that capitalization does not matter, which is not documented.

### Suggested fix for documentation

1. List supported floating point types. 
2. Mention `float16` not supported. (Could be implicitly solved by point 1.)
3. Mention entries are not case-sensitive.
4. Link to [NumPy types](https://numpy.org/doc/stable/reference/arrays.scalars.html#sized-aliases).

---

Resulting from https://github.com/pandas-dev/pandas/issues/50213.

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #50216
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Updated documentation for `pandas.to_numeric()`, pointed out that `downcast` parameter doesn’t support `float16` dtype. Listed in the description of [dtype](https://pandas.pydata.org/docs/user_guide/basics.html#dtypes) supported floating point types and added Link to [NumPy types](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy).

### CODE REVIEW

File: doc/source/user_guide/basics.rst [hunk 1]
```diff
@@ -2050,33 +2050,42 @@ The following table lists all of pandas extension types. For methods requiring `
...
-+-------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
-| :ref:`Strings <text>`                           | :class:`StringDtype`      | :class:`str`       | :class:`arrays.StringArray`   | ``'string'``                           |
-+-------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
-| :ref:`Boolean (with NA) <api.arrays.bool>`      | :class:`BooleanDtype`     | :class:`bool`      | :class:`arrays.BooleanArray`  | ``'boolean'``                          |
-+-------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
++---------------------------------------------------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
```
Comment (mroeschke) [inline doc/source/user_guide/basics.rst:2053]: Could you post a screenshot of how this looks? It's somewhat hard to see what changed from the diff

Comment (vijaybirju) [reply doc/source/user_guide/basics.rst:2053]: so, the issue is fixed? Right ! or can i still push my changes? I am new in open source

Comment (mroeschke) [reply doc/source/user_guide/basics.rst:2053]: Yes, if interested in contributing you can find an issue that hasn't been assigned yet or doesn't have an open pull request

Comment (vijaybirju) [reply doc/source/user_guide/basics.rst:2053]: Ok, thank you. So on an issue only one person can pull a request at one time?

Comment (vijaybirju) [reply doc/source/user_guide/basics.rst:2053]: Ok thanks mroeschke

File: pandas/core/tools/numeric.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def to_numeric(
 
         - 'integer' or 'signed': smallest signed int dtype (min.: np.int8)
         - 'unsigned': smallest unsigned int dtype (min.: np.uint8)
-        - 'float': smallest float dtype (min.: np.float32)
+        - 'float': float dtype (min.: np.float32), float16 is not supported.
```
Comment (mcp292) [inline pandas/core/tools/numeric.py:73]:
```suggestion
        - 'float': float dtype (min.: np.float32, float16 is not supported).
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this - I think if you just add "nullable float" to the table (without the rest), then it won't be necessary to resize the table and the diff will be legible, and then we can get this in

File: pandas/core/tools/numeric.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcp292) [reply pandas/core/tools/numeric.py:73]: This is a good point. After re-consideration, I agree.

Comment (mcp292) [reply pandas/core/tools/numeric.py:73]: See https://github.com/pandas-dev/pandas/pull/50738#discussion_r1084550181 instead.

File: doc/source/user_guide/basics.rst (hunk 1 above)
Comment (natmokval) [reply doc/source/user_guide/basics.rst:2053]: dtypes_pandas

This is a screenshot of that part of the table. 
As Marco said, there is no need to add NumPy integer and NumPy float to the table of pandas extension types. I corrected the table and left only "nullable float" under "nullable integer" (without the rest). This implies that we shouldn’t resize the table.

File: doc/source/user_guide/basics.rst [hunk 3]
```diff
@@ -2050,33 +2050,42 @@ The following table lists all of pandas extension types. For methods requiring `
...
+|                                                                                             |                           |                    |                               | ``'UInt32'``, ``'UInt64'``             |
++---------------------------------------------------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
+| `NumPy integer <https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.integer>`_ | :class:`numpy.integer`    | :class:`int`       | :class:`arrays.IntegerArray`  | ``'int8'``, ``'int16'``, ``'int32'``,  |
+|                                                                                             |                           |                    |                               | ``'int64'``, ``'uint8'``, ``'uint16'``,|
+|                                                                                             |                           |                    |                               | ``'uint32'``, ``'uint64'``             |
```
[the remark this answers is not in the document]
Comment (natmokval) [reply doc/source/user_guide/basics.rst]: Thank you, @MarcoGorelli . As you suggested, I added "nullable float" to the table and removed integer and float. 

I noticed that there is no link to “nullable float”
Wouldn’t be useful to have a description and examples for “nullable float”, as we have for [“nullable integer”](https://pandas.pydata.org/docs/user_guide/integer_na.html#integer-na)

File: doc/source/user_guide/basics.rst [hunk 4]
```diff
@@ -2050,33 +2050,35 @@ The following table lists all of pandas extension types. For methods requiring `
...
-| :ref:`Strings <text>`                           | :class:`StringDtype`      | :class:`str`       | :class:`arrays.StringArray`   | ``'string'``                           |
-+-------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
-| :ref:`Boolean (with NA) <api.arrays.bool>`      | :class:`BooleanDtype`     | :class:`bool`      | :class:`arrays.BooleanArray`  | ``'boolean'``                          |
-+-------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
++--------------------------------------------------+---------------------------+--------------------+-------------------------------+----------------------------------------+
+| Kind of Data                                     | Data Type                 | Scalar             | Array                         | String Aliases                         |
```
Comment (MarcoGorelli) [inline doc/source/user_guide/basics.rst:2054]: looks like this has gone forward by one space 😄 any chance you could bring it back to where it was, to minimise the diff?

File: pandas/core/tools/numeric.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply pandas/core/tools/numeric.py:73]: let's revert this change

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for updating! almost there

File: doc/source/user_guide/basics.rst (hunk 4 above)
Comment (natmokval) [reply doc/source/user_guide/basics.rst]: Thank you, @MarcoGorelli . I was wondering why my diff was so unreadable. I deleted one space, now it looks nicer 😄.

Also, I reverted line 73 as you suggested.

File: doc/source/user_guide/basics.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply doc/source/user_guide/basics.rst]: yeah, maybe - currently there's no user guide page on nullable floats, but perhaps there should be

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me - @phofl any objections?

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @natmokval

**Pull request merged**